File: BaseLibrary/communication/command/miner/DetonateMineCommand.cs
using System;
using System.Collections.Generic;

namespace BaseLibrary.communication.command.miner {
    public class DetonateMineCommand : AMinerCommand {

        private static readonly List<ISubCommandFactory> SUB_COMMAND_FACTORIES = new List<ISubCommandFactory>();

        public static int RegisterSubCommandFactory(ISubCommandFactory subCommandFactory) {
            int position = SUB_COMMAND_FACTORIES.Count;
            SUB_COMMAND_FACTORIES.Add(subCommandFactory);
            return position;
        }

        private int _mineID;
        /// <summary>
        /// ID of mine which want to detonate.
        /// </summary>
        public int MINE_ID {
            get {
                if (pending)
                    throw new NotSupportedException("Cannot access to property of pending request.");
                return _mineID;
            } private set => _mineID = value; }

        public DetonateMineCommand(int mineId) {
            MINE_ID = mineId;
        }
    }
}

File: Java-Client/RobotBattlefield-Client-Java/src/BaseLibrary/communication/command/miner/DetonateMineCommand.java

package BaseLibrary.communication.command.miner;

import BaseLibrary.communication.protocol.AProtocol;

public class DetonateMineCommand extends AMinerCommand {

	private static final String NAME = "DETONATE_MINE";

	static {
		AProtocol.registerForDeserialize(NAME, DetonateMineCommand.class);
	}

	@Override
	public String getCommandName() {
		return NAME;
	}
	
	private int mineID;

	public DetonateMineCommand(int mineID) {
		setMineID(mineID);
	}
	
	/**
	 * ID of mine which want to detonate.
	 */
	public int getMineID() {
		if (pending)
			throw new UnsupportedOperationException("Cannot access to property of pending request.");
		return mineID;
	}

	public void setMineID(int mineID) {
		this.mineID = mineID;
		pending = false;
	}

}
